src/meta/MetaHttpHandler.cpp

```diff
@@ -139,5 +196,66 @@ std::string MetaHttpHandler::toStr(folly::dynamic& vals) const {
     return ss.str();
 }
 
+bool MetaHttpHandler::dispatchSSTFiles(const std::string& url,
+                                       int port,
+                                       const std::string& path,
+                                       const std::string& local) {
+    auto command = folly::stringPrintf("hdfs dfs -ls hdfs://%s:%d%s ",
+                                       url.c_str(), port, path.c_str());
+    LOG(INFO) << "List SST Files " << command;
+    auto result = ProcessUtils::runCommand(command.c_str());
+    std::vector<std::string> files;
+    folly::split("\n", result.value(), files, true);
+    auto partNumber = std::count_if(files.begin(), files.end(), [](const std::string& element) {
+                                        return element.find("part") != std::string::npos;
+                                    });
+
+    auto partResult = (*metaClient).getPartsAlloc(spaceID).get();
+    if (!partResult.ok()) {
+        return false;
+    }
+
+    if (folly::to<int64_t>(partNumber) != (*metaClient).partsNum(spaceID)) {
+        LOG(ERROR) << "HDFS part number should be equal with nebula";
+        return false;
+    }
+
+    auto hosts = (*metaClient).listHosts().get();
+    if (!hosts.ok()) {
+        LOG(ERROR) << "List Hosts Failed: " << hosts.status();
+        return false;
+    }
+
+    for (auto iter = hosts.value().cbegin(); iter != hosts.value().cend(); iter++) {
+        auto partMap = (*metaClient).getPartsMapFromCache((*iter));
+        auto id = partMap.find(spaceID);
+        if (id == partMap.end()) {
+            LOG(ERROR) << "Can't find SpaceID " << spaceID;
+            return false;
+        }
+        std::vector<PartitionID> partitions;
+        auto partIter = partMap[spaceID].begin();
+        while (partIter != partMap[spaceID].end()) {
+            partitions.emplace_back(partIter->first-1);
+            partIter++;
+        }
+        std::string parts;
+        folly::join(",", partitions, parts);
+
+        auto host = network::NetworkUtils::intToIPv4((*iter).first);
```

Comment:
Parameter type mismatch between 'uint32_t' and 'IPv4'

Reply:
Good catch  👍  change to  `int32_t`